src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1280,40 +1285,13 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                 _cancelToken = new CancellationTokenSource();
                 try
                 {
-                    long requestContentLength = request.Content is null ? 0 : request.Content.Headers.ContentLength.Value;
-
-                    string reqVerboseMsg = string.Format(
-                        CultureInfo.CurrentCulture,
-                        WebCmdletStrings.WebMethodInvocationVerboseMsg,
-                        request.Version,
-                        request.Method,
-                        requestContentLength);
-
-                    WriteVerbose(reqVerboseMsg);
-
-                    string reqDebugMsg = string.Format(
-                        CultureInfo.CurrentCulture,
-                        WebCmdletStrings.WebRequestDebugMsg,
-                        request.ToString());
-
-                    WriteDebug(reqDebugMsg);
+                    WriteMessageIfPreferenceEnabled(() => GetWebRequestVerboseInfoString(request), isDebug: false);
+                    WriteMessageIfPreferenceEnabled(() => GetWebRequestDebugInfoString(request, InFile is not null, _cancelToken.Token), isDebug: true);
 
                     response = client.SendAsync(currentRequest, HttpCompletionOption.ResponseHeadersRead, _cancelToken.Token).GetAwaiter().GetResult();
 
-                    string contentType = ContentHelper.GetContentType(response);
-                    long? contentLength = response.Content.Headers.ContentLength;
-                    string respVerboseMsg = contentLength is null
-                        ? string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseNoSizeVerboseMsg, response.Version, contentType)
-                        : string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseVerboseMsg, response.Version, contentLength, contentType);
-
-                    WriteVerbose(respVerboseMsg);
-
-                    string resDebugMsg = string.Format(
-                        CultureInfo.CurrentCulture,
-                        WebCmdletStrings.WebResponseDebugMsg,
-                        response.ToString());
-
-                    WriteDebug(resDebugMsg);
+                    WriteMessageIfPreferenceEnabled(() => GetWebResponseVerboseInfoString(response), isDebug: false);
+                    WriteMessageIfPreferenceEnabled(() => GetWebResponseDebugInfoString(response, _cancelToken.Token), isDebug: true);
```

Comment:
Oh, I don't want us to create these delegates on the fly. Earlier, you convinced me that static is not worth it.

Reply:
Wasn't sure if I did. I'll revert it.

Reply:
Harder than I thought to deduplicate that and not have it be a mess so will full revert.